File: BACKEND/excepciones13062022/src/excepciones13062022/Piscina.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package excepciones13062022;

/**
 *
 * @author Gaby
 */
public class Piscina {

    private int nivel;
    public final int MAX_NIVEL;

    public Piscina(int max) {
        if (max < 0) {
            max = 0;
        }
        MAX_NIVEL = max;
    }

    public int getNivel() {
        return nivel;
    }

    public void vaciar(int cantidad) throws Exception {

        if (nivel - cantidad < 0) {
            throw new Exception();
        } else {
            nivel = nivel - cantidad;
        }

    }
public void llenar(int cantidad) throws Exception {

        if (nivel + cantidad < MAX_NIVEL) {
            throw new Exception();
        } else {
            nivel = nivel + cantidad;
        }

    }
   
}

File: BackEnd/manejoErrores/manejoErrores/Piscina.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace manejoErrores
{
    class Piscina
    {
        private int nivel;
        public  int MAX_NIVEL=0;
        public Piscina(int max)
        {
            if (max < 0) max = 0;
            MAX_NIVEL = max;
        }
        public int getNivel()
        {
            return nivel;
        }

        public void vaciar (int cantidad)
        {
            if (cantidad > nivel)
            {
                //throw new Exception("No se puede vaciar la piscina");
                throw new controlngresoException($"la cantidad {cantidad} es mayor que el nivel {nivel}");

            }
   
                nivel = nivel - cantidad;
              

            
        }

        public void llenar(int cantidad)
        {
            if (cantidad > MAX_NIVEL)
            {
                throw new Exception("El llenado no puede superar a {MAX_NIVEL}");
            }
            nivel = nivel + cantidad;
        }


    }
}
